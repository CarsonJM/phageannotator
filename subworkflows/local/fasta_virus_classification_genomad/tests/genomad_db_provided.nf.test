nextflow_workflow {

    name "Test Subworkflow: FASTA_VIRUS_CLASSIFICATION_GENOMAD"
    script "subworkflows/local/fasta_virus_classification_genomad/main.nf"
    workflow "FASTA_VIRUS_CLASSIFICATION_GENOMAD"
    tag "subworkflows_local"
    tag "fasta_virus_classification_genomad"
    tag "genomad_db_provided"


    test("Test --genomad_db provided") {
        when {
            params {
                outdir      = "$outputDir"
            }
            workflow {
                """
                input[0] = Channel.of([ [ id:'test' ], [ file('https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true) ] ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert workflow.trace.succeeded().size() == 2 },
                { assert snapshot(path("$outputDir/VirusClassification//").list()).match("fasta_virus_classification_genomad") }
            )
        }
    }
}
