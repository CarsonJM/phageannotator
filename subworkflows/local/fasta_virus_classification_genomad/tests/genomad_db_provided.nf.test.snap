{
    "fasta_virus_classification_genomad": {
        "content": [
            [
                [
                    [
                        "genome_virus.fna.gz:md5,c2d03e27ae27e3eedd037621ee77c917"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-09-13T17:54:06+0000"
    }
}